File: ClientApp/src/app/shared/Models/serviceline.ts
import { JsonApiModelConfig, JsonApiModel, Attribute, HasMany, BelongsTo } from 'angular2-jsonapi';
import { BaseTitleObject, BaseObject } from './_baseTitleObject';
import { Wsshour } from './wsshour';
import { Service } from './service';
import { Project } from './project';
import { Client } from './client';
import { Tabel } from './tabel';

@JsonApiModelConfig({ type: 'servicelines' })
export class Serviceline extends BaseObject {
  @BelongsTo()  service: Service;
  @BelongsTo()  project: Project;
  @BelongsTo()  client: Client;
  @BelongsTo()  tabel: Tabel;
  @Attribute()  result: string;
  @HasMany() hours: Wsshour[];
}

File: Models/DB/ServiceLine.cs
using JsonApiDotNetCore.Models;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace SURV.Models.DB
{
    public class Serviceline : BaseObjectSoftDelete
    {
        [ForeignKey(nameof(Service))] public int? ServiceId { get; set; }

        [HasOne, Required] public virtual Service Service { get; set; }

        [ForeignKey(nameof(Project))] public int? ProjectId { get; set; }
        [HasOne, Required] public virtual Project Project { get; set; }

        [ForeignKey(nameof(Client))] public int? ClientId { get; set; }
        [HasOne, Required] public virtual Client Client { get; set; }

        [ForeignKey(nameof(Tabel))] public int? TabelId { get; set; }
        [HasOne, Required] public virtual Tabel Tabel { get; set; }

        [Required] [Attr("result")] public string Result { get; set; }

        [HasMany] public virtual List<Wsshour> Hours { get; set; }
    }
}
